lib/inertia_rails/inertia_rails.rb

```diff
@@ -1,52 +1,41 @@
 # Needed for `thread_mattr_accessor`
 require 'active_support/core_ext/module/attribute_accessors_per_thread'
-require 'inertia_rails/lazy'
+require 'inertia_rails/mergeable_prop'
+require 'inertia_rails/base_prop'
+require 'inertia_rails/ignore_first_load_prop'
+require 'inertia_rails/optional_prop'
+require 'inertia_rails/always_prop'
+require 'inertia_rails/merge_prop'
+require 'inertia_rails/defer_prop'
+require 'inertia_rails/configuration'
 
 module InertiaRails
-  def self.configure
-    yield(Configuration)
-  end
-
-  def self.version
-    Configuration.evaluated_version
-  end
-
-  def self.layout
-    Configuration.layout
-  end
-
-  def self.ssr_enabled?
-    Configuration.ssr_enabled
-  end
-
-  def self.ssr_url
-    Configuration.ssr_url
-  end
+  class << self
+    CONFIGURATION = Configuration.default
 
-  def self.default_render?
-    Configuration.default_render
-  end
+    def configure
+      yield(CONFIGURATION)
+    end
 
-  def self.deep_merge_shared_data?
-    Configuration.deep_merge_shared_data
-  end
+    def configuration
+      CONFIGURATION
+    end
 
-  def self.lazy(value = nil, &block)
-    InertiaRails::Lazy.new(value, &block)
-  end
+    def optional(value = nil, &block)
+      OptionalProp.new(value, &block)
+    end
+    alias_method :lazy, :optional
```

Comment:
I think lazy will be deprecated, we can do that with alias?

Reply:
Added the deprecation, thanks!